coeus-impl/src/main/java/org/kuali/coeus/common/budget/impl/rate/BudgetRatesServiceImpl.java

```diff
@@ -173,6 +179,8 @@ protected void checkActivityPrefixForRateClassTypes(List<RateClassType> rateClas
         List<BudgetLaRate> budgetLaRates = budget.getBudgetLaRates();
         for(RateClassType rateClassType : rateClassTypes) {
             if(rateClassType.getPrefixActivityType()) {
+            	//making changes to the DO here, need to detach to make sure these changes aren't persisted.
+            	entityManager.detach(rateClassType);
                 String newRateClassTypeDescription = activityTypeDescription.concat(rateClassType.getDescription()); 
                 rateClassType.setDescription(newRateClassTypeDescription);
                 rateClassType.setPrefixActivityType(false);
```

Comment:
No. Definitely not. The rateClassType is maint doc managed. It has a flag on it, prefixActivityType and if that is set then the description should be prefixed the with current budget activity type in the view, but if we persist that then the first budget created will modify that maint doc record and all future budgets will have that activity type for that rate class type.

Would be more correct to use a DTO here instead anyway, but this code is very complex and I didn't want to tweak it more than necessary.